src/apis/logs_api.ts

```diff
@@ -43,4 +134,56 @@ export class LogsAPI {
 
         return parser.parseDecodedLogs(decodedLogs);
     }
+
+    /**
+     * TODO: Add watch function.
+     *
+     * @example
+     * // Get a listener for "LogLoanOrderFilled" events
+     * dharma.logs.watch("LogLoanOrderFilled");
+     *
+     * @param {string | string[]} eventNames
+     * @returns {Promise<any>}
+     *
+     * public watch(eventNames: string | string[]): Promise<any>;
+     */
+
+    private async getEventToContractsMap() {
+        const contractWrappers = await this.getContractWrapper();
+
+        contractWrappers.map((wrapper) => ABIDecoder.addABI(wrapper.abi));
+
+        // Create a mapping of event names to the contract where they originate.
```

Comment:
Instead of this comment, I would separate this section into a separate function that takes in wrapper and spits out an object. That way the logic in `getEventToContractsMap` is cleaner.

Reply:
i think this would look cleaner if there was a clear chain of transformations